modules/substrate/src/justification.rs

```diff
@@ -230,26 +207,44 @@ mod tests {
 		}
 	}
 
-	fn make_justification_for_header_1() -> GrandpaJustification<TestHeader> {
+	pub(crate) fn make_justification_for_header(
+		header_index: u8,
+		round: u64,
+		set_id: SetId,
+		authorities: &AuthorityList,
+	) -> GrandpaJustification<TestHeader> {
+		let (target_hash, target_number) = header_id(header_index);
+		let mut precommits = vec![];
+		let mut votes_ancestries = vec![];
+
+		// I'm using the same header for all the voters since it doesn't matter as long
+		// as they all vote on blocks _ahead_ of the one we're interested in finalizing
+		for (id, _weight) in authorities {
+			let signer = extract_keyring(&id);
+			let precommit = signed_precommit(signer, header_id(header_index + 1), round, set_id);
+			precommits.push(precommit);
+			votes_ancestries.push(test_header((header_index + 1).into()));
```

Comment:
You never change `header_index` => are you sure that you need to duplicate same header in `votes_ancestries` 3 times? Not an issue, though.

Reply:
I was under the impression that we needed one header per authority for the vote ancestries. Looks like the tests pass if I just have one header in there though. I'll need to think about what more closely matches what Grandpa does